### PR TITLE
Make some tweaks to support clang-cl for visual studio cleanly

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will make a few tweaks generated from warnings when building HELICS with clang-cl through Visual studio

### CODE REVIEW

File: src/helics/core/helicsTime.hpp [hunk 1]
```diff
@@ -39,12 +39,6 @@ constexpr Time cBigTime = Time{static_cast<int64_t>(HELICS_BIG_NUMBER * 1000000)
 /** common definition of currentTime in initialization mode*/
 constexpr Time initializationTime = negEpsilon;
 
-/** user defined literal for a time variable*/
-constexpr Time operator"" _t(long double val)
-{
-    return {static_cast<double>(val)};
-}  // NOLINT
-
```
Comment (nightlark) [inline src/helics/core/helicsTime.hpp:47]: Is this function part of the public API?

File: src/helics/core/CoreBroker.cpp [hunk 2]
```diff
@@ -2573,16 +2573,16 @@ static InterfaceType getMatchType(InterfaceType type)
...
     const BasicHandleInfo& origin,
+    uint32_t originFlags,
     const BasicHandleInfo& target,
-    uint32_t flagsSource,
-    uint32_t flagsDest,
+    uint32_t targetFlags,
     std::pair<action_message_def::action_t, action_message_def::action_t> actions)
 {
-    // notify the target about a source
+    // notify the target about a origin
```
Comment (nightlark) [inline src/helics/core/CoreBroker.cpp:2581]:
```suggestion
    // notify the target about an origin
```

File: src/helics/application_api/Subscriptions.hpp [hunk 3]
```diff
@@ -20,33 +20,6 @@ SPDX-License-Identifier: BSD-3-Clause
...
-@param fed a reference to a valueFederate
-@param key  the name of the publication
-@param obj the obj to store the retrieved value
-*/
-template<class X>
-void getValue(ValueFederate& fed, std::string_view key, X& obj)
-{
-    fed.getSubscription(key).getValue<X>(obj);
-}
-
```
Comment (nightlark) [inline src/helics/application_api/Subscriptions.hpp:49]: If these are part of the public API, we should update the changelog to note that they are gone now and what alternative function to use.

Comment (phlptp) [reply src/helics/application_api/Subscriptions.hpp:49]: The C++ API only,  so yes it was public but also not used as far as I can tell, and generated some deprecation warnings and I think they have nearly completely been superceeded by other operations.  And I agree the changelog will need to be updated.

File: src/helics/core/helicsTime.hpp (hunk 1 above)
Comment (phlptp) [reply src/helics/core/helicsTime.hpp:47]: yes,  though like the above generated several warnings, is not very useful, and was generally confusing and not used other than a test case which I changed.

Comment (phlptp) [reply src/helics/core/helicsTime.hpp:47]: but a changelog note is warranted.

**Pull request merged**